File: src/easy/majority-element.ts
// https://edabit.com/challenge/aewSLr2r2LMSDhPxf
// https://leetcode.com/problems/majority-element/
export const majorityElement = (
    nums: Array<number | string>,
): number | null => {
    const half = nums.length / 2;
    const freq = new Map();
    for (const num of nums) {
        freq.set(num, (freq.get(num) ?? 0) + 1);
    }
    for (const [key, value] of freq.entries()) {
        if (value > half) return key;
    }
    return null;
};

File: lib/easy/majority_element.dart
int majorityElement(List<int> nums) {
  int count = 0;
  int? candidate;

  for (final value in nums) {
    if (count == 0) {
      candidate = value;
    }
    count += (value == candidate) ? 1 : -1;
  }
  return candidate ?? 0;
}
